tmt/steps/finish/__init__.py

```diff
@@ -29,7 +29,16 @@ def go(self):
         """ Prepare the test step """
         super().go()
 
-        # Stop and remove provisioned guests
-        for guest in self.plan.provision.guests():
-            guest.stop()
-            guest.remove()
+        available_guests = self.plan.provision.guests()
+        if available_guests and self.opt('keep'):
+            self.info("Guest will keep running", shift=1)
+            self.info("Run `tmt run --id {} finish` later to finish properly".format(self.parent.run.workdir), shift=1)
+        for guest in available_guests:
+            if self.opt('keep'):
+                # Print details about connection
+                self.info("guest", guest.name, color='magenta', shift=1)
+                self.info('connect', "ssh {} {}".format(guest._ssh_guest(), guest._ssh_options(True)), 'green', shift=1)
```

Comment:
I think this part should go to the `Guest` class. We could introduce a new `keep()` method which would serve this information and allow inheriting classes to override if necessary. This will be useful for example for the `podman` plugin which can provide its own hint how to connect to the container.